src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalysisScope.cs

```diff
@@ -152,11 +175,15 @@ public bool ShouldAnalyze(ISymbol symbol)
                 return false;
             }
 
-            foreach (var location in symbol.Locations)
+            foreach (var syntaxRef in symbolEvent.DeclaringSyntaxReferences)
             {
-                if (FilterFileOpt.Value.SourceTree == location.SourceTree && ShouldInclude(location.SourceSpan))
+                if (syntaxRef.SyntaxTree == FilterFileOpt.Value.SourceTree)
                 {
-                    return true;
+                    var node = getTopmostNodeForAnalysis(symbolEvent.Symbol, syntaxRef, symbolEvent.Compilation, cancellationToken);
```

Comment:
This fixes the issue mentioned in point 1. of the PR comment

> Symbol/SyntaxNode/Operation analyzer - execute callbacks only on symbols/nodes/operations whose span intersects with the input filter span. The driver already did this prior to this PR, but needed a small bug fix for symbol action callbacks to consider the entire symbol declaration span, not just the symbol's location span (which only contains the span for the symbol header).

Reply:
i'm trying ot figure out how this works.  we're iterating through teh references (which may be in seaprate files), but we're passing the node's span *alone* below.  How do you know the span refers to the document with the line you're caring about?  can you doc what's doing on here, and why it's ok to just look at spans from nodes from disparate files?

Reply:
> How do you know the span refers to the document with the line you're caring about?

We have a check just above that ensures we only process SyntaxRefs whose SyntaxTree matches the SourceTree of the FilterFile.

Reply:
d'oh. totally misssed that :)